### PR TITLE
Simplify await_up logic.

### PR DESCRIPTION
**Pull request by jchristgit**
Gun 2.0 allows us to do this.

### CODE REVIEW

File: lib/nostrum/shard/session.ex [hunk 1]
```diff
@@ -61,7 +61,7 @@ defmodule Nostrum.Shard.Session do
     {:ok, worker} = :gun.open(:binary.bin_to_list(gateway), 443, @gun_opts)
     {:ok, :http} = :gun.await_up(worker, @timeout_connect)
     stream = :gun.ws_upgrade(worker, @gateway_qs)
-    await_ws_upgrade(worker, stream)
+    {:upgrade, ["websocket"], _} = :gun.await(worker, stream, @timeout_ws_upgrade)
```
Comment (Kraigie) [inline lib/nostrum/shard/session.ex:64]: Are there alternative return values for this function now we should handle? Previously we had a whole section of code that provided detailed errors when things went awry.

Comment (jchristgit) [reply lib/nostrum/shard/session.ex:64]: Only `:gun_error` and a timeout, which will cause an exit with badmatch.

**Review by jb3 — APPROVED** [review]
Tested & works fine, good stuff.

Only comment is that instead of exiting out with badmatches we might want to try catch those and reshape them into descriptive nostrum errors about failures with Discord Gateway, but this is optional (no matter what the error is it'd be fatal).

**Comment by jchristgit** [comment]
@Kraigie How do we continue here? Do we close, since we will likely have to go back to gun 2.0 to allow making a new release, or do we merge?

**Comment by jchristgit** [comment]
I am closing this for now. The simplification is nice, but we don't want to block a new nostrum release on a new `:gun` release.

**Pull request closed without merging**